src/utils.cpp

```diff
@@ -604,62 +604,46 @@ namespace, it will throw an exception
 */
 static bool pid_in_different_mountns(int pid)
 {
-
-  struct stat self_stat, target_stat;
-  int self_fd = -1, target_fd = -1;
-  std::stringstream errmsg;
-  char buf[64];
-
   if (pid <= 0)
     return false;
 
-  if ((size_t)snprintf(buf, sizeof(buf), "/proc/%d/ns/mnt", pid) >= sizeof(buf))
-  {
-    errmsg << "Reading mountNS would overflow buffer.";
-    goto error;
-  }
+  std::error_code ec;
+  std_filesystem::path self_path{ "/proc/self/ns/mnt" };
+  std_filesystem::path target_path{ "/proc" };
+  target_path /= std::to_string(pid);
+  target_path /= "ns/mnt";
 
-  self_fd = open("/proc/self/ns/mnt", O_RDONLY);
-  if (self_fd < 0)
+  if (!std_filesystem::exists(self_path, ec))
   {
-    errmsg << "open(/proc/self/ns/mnt): " << strerror(errno);
-    goto error;
+    throw MountNSException(
+        "Failed to compare mount ns with PID " + std::to_string(pid) +
+        ". The error was open (/proc/self/ns/mnt): " + ec.message());
   }
 
-  target_fd = open(buf, O_RDONLY);
-  if (target_fd < 0)
+  if (!std_filesystem::exists(target_path, ec))
   {
-    errmsg << "open(/proc/<pid>/ns/mnt): " << strerror(errno);
-    goto error;
+    throw MountNSException(
+        "Failed to compare mount ns with PID " + std::to_string(pid) +
+        ". The error was open (/proc/<pid>/ns/mnt): " + ec.message());
   }
 
-  if (fstat(self_fd, &self_stat))
+  if (target_path.string().length() >= 64)
   {
-    errmsg << "fstat(self_fd): " << strerror(errno);
-    goto error;
+    throw MountNSException(
+        "Failed to compare mount ns with PID " + std::to_string(pid) +
+        ". The error was Reading mountNS would overflow buffer.");
```

Comment:
Do we still need this check? Isn't the 64 length limit b/c of `char buf[64]` which is deleted now?

Reply:
@danobi Hi, just want to be backward-compatible, so the same exceptions are thrown

Reply:
@danobi Hi, I have removed the overflow exception

Reply:
> Hi, just want to be backward-compatible, so the same exceptions are thrown

I don't think anyone will mind if there are less errors than before :)